torch/csrc/jit/codegen/cuda/executor_utils.cpp

```diff
@@ -370,7 +372,40 @@ bool canVectorize(const IValue& aten_val, int word_size) {
   return true;
 }
 
-namespace {
+// Returns true if a TV can be used with ParallelType::Vectorize. When
+// input or output tensors are involved, the other version of
+// canVectorize is used.
+bool canVectorize(
+    TensorView* tv,
+    int word_size,
+    kir::ExpressionEvaluator& expr_eval) {
+  IterDomain* last_root_dim = nullptr;
+  for (size_t i = tv->getRootDomain().size(); i > 0; i--) {
+    auto r_id = tv->getRootDomain()[i - 1];
+    if (r_id->isReduction() || r_id->isTrivialReduction() ||
+        r_id->isBroadcast()) {
+      continue;
+    }
+    last_root_dim = r_id;
+    break;
+  }
+
+  if (last_root_dim == nullptr) {
+    return false;
+  }
+
+  auto last_dim_size = expr_eval.evaluate(last_root_dim->extent());
```

Comment:
This path should be precomputed so expr eval should be just one look up. Note to myself to double check for intermediate tensors.